src/libsyntax_ext/asm.rs

```diff
@@ -67,6 +68,19 @@ pub fn expand_asm<'cx>(cx: &'cx mut ExtCtxt<'_>,
     }))
 }
 
+fn parse_asm_str<'a>(p: &mut Parser<'a>) -> PResult<'a, Symbol> {
+    match p.parse_str_lit() {
+        Ok(str_lit) => Ok(str_lit.symbol_unescaped),
+        Err(opt_lit) => {
+            let span = opt_lit.map_or(p.token.span, |lit| lit.span);
+            let msg = "expected string literal";
+            let mut err = p.sess.span_diagnostic.struct_span_fatal(span, msg);
```

Comment:
Should this be fatal? Could we error here, return `Option<Symbol>` instead, and just expand the ASM to `ExprKind::Err`? (If you think that's low priority I would agree, but I thought it would be worth noting.)